numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -2369,6 +2471,50 @@ array_vdot(PyObject *NPY_UNUSED(dummy), PyObject *args)
 }
 
 
+NPY_NO_EXPORT PyObject *
+array_matmul(PyObject *NPY_UNUSED(m), PyObject *args, PyObject* kwds)
+{
+    static PyUFuncObject *cached_npy_matmul = NULL;
+    int errval;
+    PyObject *override = NULL;
+    PyObject *v, *a, *o = NULL;
+    char* kwlist[] = {"a", "b", "out", NULL };
+
+    if (cached_npy_matmul == NULL) {
+        PyObject *module, *dict, *matmul;
+
+        module = PyImport_ImportModule("numpy.core.multiarray");
+        dict = PyModule_GetDict(module);
+        matmul = PyDict_GetItemString(dict, "matmul");
+        cached_npy_matmul = (PyUFuncObject*)matmul;
+        Py_INCREF(cached_npy_matmul);
+        Py_DECREF(module);
+    }
+
+    errval = PyUFunc_CheckOverride(cached_npy_matmul, "__call__",
+                                   args, kwds, &override, 2);
```

Comment:
I feel like `PyArray_Matmul` should also do the override check, i.e. it should have identical semantics to the `matmul` function?